src/lib/holocene/toggle-switch.svelte

```diff
@@ -1,53 +1,130 @@
+<!-- Based loosely on https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/a -->
 <script lang="ts">
   export let checked = false;
   export let id: string;
   export let disabled = false;
+  export let label: string;
+  export let labelPosition: 'left' | 'right' = 'right';
+  export let labelHidden = false;
+
+  let focused = false;
+  let hovered = false;
+
+  const handleFocus = () => {
+    focused = true;
+  };
+
+  const handleBlur = () => {
+    focused = false;
+  };
+
+  const handleMouseEnter = () => {
+    hovered = true;
+  };
+
+  const handleMouseLeave = () => {
+    hovered = false;
+  };
 </script>
 
-<label for={id} class="switch">
+<label class="switch" class:disabled>
+  {#if labelPosition === 'left'}
+    <span class="label left" class:sr-only={labelHidden}>
+      {label}
+    </span>
+  {/if}
   <input
     on:change
     {id}
     {disabled}
     bind:checked
     type="checkbox"
-    class="hidden"
-    class:disabled
+    class="checkbox"
+    role="switch"
+    on:focus={handleFocus}
+    on:blur={handleBlur}
     data-testid={$$props['data-testid']}
   />
-  <span class="slider" />
+  <span
+    on:mouseenter={handleMouseEnter}
+    on:mouseleave={handleMouseLeave}
+    class="slider"
+    class:disabled
+    class:checked
+    class:focused
+    class:hovered
+  >
+    <span class="thumb" class:checked />
+  </span>
+  {#if labelPosition === 'right'}
+    <span class="label right" class:sr-only={labelHidden}>
+      {label}
+    </span>
+  {/if}
 </label>
 
 <style lang="postcss">
   .switch {
-    @apply relative inline-block h-8 w-[52px] rounded-[8rem];
+    @apply px-2 relative flex items-center rounded max-w-fit cursor-pointer;
```

Comment:
We might want a `min-w-fit` here instead

```suggestion
    @apply px-2 relative flex items-center rounded min-w-fit cursor-pointer;
```

so that it doesn't get messed up on smaller screens

| Before | After |
|--|--|
|<img width="320" alt="Screenshot 2023-08-02 at 9 30 24 AM" src="https://github.com/temporalio/ui/assets/15069288/4d940ea9-17f4-47eb-902d-fa53f0791bfd">|<img width="315" alt="Screenshot 2023-08-02 at 9 31 07 AM" src="https://github.com/temporalio/ui/assets/15069288/7e6a5fde-1768-4d6b-b34c-03b4b52193de">|